walker.go

```diff
@@ -161,6 +162,14 @@ func Walk(ctx context.Context, p string, opt *WalkOpt, fn filepath.WalkFunc) err
 			}
 
 			if isDir {
+				fi, err := dirEntry.Info()
+				if err != nil {
+					if isNotExist(err) {
+						return nil
```

Comment:
this should be `filepath.SkipDir`?

Reply:
If this is the case, then you can just remove this condition as `SkipDir` is handled in the defer funciton.

Reply:
ya, i think that would be slightly better

notice that this is handling the narrow case where Readdir() saw a dir, then the dir was removed before we had a chance to walk it.